### ISSUE TITLE
Variables not looping correcly?

### ISSUE THREAD

**Issue by MalfuncEddie** [grafana/grafana-polystat-panel#128]
Hi I am trying to implement a status panel with multiple stats 
I've got a PostgreSQL query that returns following data
"time","name","Sev4","Sev3","Sev2","Sev1"
"2020-05-08 19:45:43.486036","VM1","0","1","0","0"
"2020-05-08 19:45:43.486036","VM2","0","0","0","0"
"2020-05-08 19:45:43.486036","VM3","0","0","0","0"

$VM is a list of the names

I want a hexagon per name/VM that has the values and override thresholds of Sev4,Sev3,Sev2,Sev1

Step 1: Create a composite:
Composite Name: $VM
Metric: $VM Sev1 Alias: Sev1
Metric: $VM Sev2 Alias: Sev2
Metric: $VM Sev3 Alias: Sev3
Metric: $VM Sev4 Alias: Sev4

Step 2:
Override 1: Name or regex $VM Sev1  
Override 2: Name or regex $VM Sev2
Override 3: Name or regex $VM Sev3
Override 4: Name or regex $VM Sev4

Expected result:
3 hexagons (VM1,VM2,VM3) with 1 in warning (VM1)

Result
3 hexagons  (VM1,VM2,VM3) all in warning
The values in the hexagons seems to match. But when I hover over the hexagon each one has Sev3 = 1

What am I doing wrong or is this a bug?

**Comment by MalfuncEddie** [comment]
It looks like composite name $VM is looping correctly but the regex metric is not correct

When I change the regex to $VM.* it takes all the values of $VM instead of looping it per composite.

**Comment by briangann** [comment]
will take a closer look at this - composite variable selection should be more flexible than it is currently.

**Comment by briangann** [comment]
a new feature is in progress that makes composites much easier to define, and should solve this issue as a side-effect.

you will be able to use a template variable as the composite name, and it will auto-match correctly. configuring 100's of composites isn't practical, i've run into the exact same issue

**Comment by briangann** [comment]
> Was anyone every able to figure out a workaround for this or is there a fix planned in the near future?
> 
> I have almost 1000 different composites using dashboard variables that all broke after upgrading (forced to upgrade for compatibility with Grafana 9).
> 
> Hoping I dont have to manually create them all.

did it break in v2? upgrading shouldn't have changed anything from v1 to v2 (and migrations were automated).

or was this a grafana v8->v9 break?

If you have a sample panel.json/dashboard to share i can work on sorting it out.  In the meantime i'll put up some examples with composited variables and metric grouping

**Comment by blauter** [comment]
Thanks for replying so quick. All my polystat panels using composites with variables were working with Grafana 8 and plugin version 1.2.7. I first upgraded from Grafana v8 -> v9 and I noticed all my polystat panels were not displaying at all so I upgraded  the plugin to 2.0 since I saw there were some changes in 1.2.9 and later to support Grafana v9. 

Still none of my panels were working so I tried to re-create a more simple one from scratch.


![2022-10-06_12-51-04](https://user-images.githubusercontent.com/18193005/194372785-a61f5b30-e289-4f32-9a80-428d7ee1a38f.gif)

Here is quick example. There is dashboard variable that contains all possible device types, for example "apacheag". In previous version, I would name my composite $device_type and then for metrics, add pattern below which matches my metrics I am retrieving from prometheus:
snowflake_${device_type}_prod Total Tasks
snowflake_${device_type}_prod Failed Tasks

With the all option selected, it does not work. However, if I select one of the device types from the drop down, it works, but for that selection only:
![2022-10-06_12-51-28](https://user-images.githubusercontent.com/18193005/194373323-de1ef230-de26-4ce8-8272-4d718b7ab749.gif)

The same configuration under previous versions of Grafana/Polystat, it wouldn't matter my selection, it always matched all based on those settings.

Hope that is clear.

**Comment by briangann** [comment]
if you select two device_types from the menu does it work, or only with one?

i'll duplicate this and get a fix out asap - this scenario is exactly what the templated composite variables are supposed to solve.  i think it's the "all" output that is throwing off the code, but it's possible multi-select is doing the same thing now as "all"

**Comment by blauter** [comment]
If I change to multiselect and select two, neither work, similar to all option.

**Comment by briangann** [comment]
v2.0.1 is being released that contains fixes for this scenario. Let us know if you run into any issues or if it resolves your problem.

Thanks!

**Comment by blauter** [comment]
Success! I had to change composite name from $device_type to ${device_type} to make it work, but it works!

I did have to re-create the panel though. The existing panel was not showing up, even after clearing cache and performing hard reload.  It did eventually re-appear but in edit mode, I was getting just black screen.

Thanks a lot for the quick fix. Worst case, I may have to re-create my existing panels but not as big of deal of manually creating 1000 composites :)
